arkouda/pdarraysetops.py

```diff
@@ -436,26 +516,64 @@ def setdiff1d(pda1: pdarray, pda2: pdarray,
     >>> b = ak.array([3, 4, 5, 6])
     >>> ak.setdiff1d(a, b)
     array([1, 2])
+
+    #Multi-Array Example
+    >>> a = ak.arange(5)
+    >>> b = ak.array([1, 5, 3, 4, 2])
+    >>> c = ak.array([1, 4, 3, 2, 5])
+    >>> d = ak.array([1, 2, 3, 5, 4])
+    >>> multia = [a, a, a]
+    >>> multib = [b, c, d]
+    >>> ak.setdiff1d(multia, multib)
+    [array([2, 4, 5]), array([2, 4, 5]), array([2, 4, 5])]
     """
-    if pda1.size == 0:
-        return pda1  # return a zero length pdarray
-    if pda2.size == 0:
-        return pda1  # subtracting nothing return orig pdarray
-    if (pda1.dtype == int and pda2.dtype == int) or \
-            (pda1.dtype == akuint64 and pda2.dtype == akuint64):
-        repMsg = generic_msg(cmd="setdiff1d", args="{} {} {}". \
-                             format(pda1.name, pda2.name, assume_unique))
-        return create_pdarray(cast(str, repMsg))
-    if not assume_unique:
-        pda1 = cast(pdarray, unique(pda1))
-        pda2 = cast(pdarray, unique(pda2))
-    return pda1[in1d(pda1, pda2, invert=True)]
+    if isinstance(pda1, pdarray) and isinstance(pda2, pdarray):
+        if pda1.size == 0:
+            return pda1  # return a zero length pdarray
+        if pda2.size == 0:
+            return pda1  # subtracting nothing return orig pdarray
+        if (pda1.dtype == int and pda2.dtype == int) or \
+                (pda1.dtype == akuint64 and pda2.dtype == akuint64):
+            repMsg = generic_msg(cmd="setdiff1d", args="{} {} {}". \
+                                 format(pda1.name, pda2.name, assume_unique))
+            return create_pdarray(cast(str, repMsg))
+        if not assume_unique:
+            pda1 = cast(pdarray, unique(pda1))
+            pda2 = cast(pdarray, unique(pda2))
+        return pda1[in1d(pda1, pda2, invert=True)]
+    elif isinstance(pda1, list) and isinstance(pda2, list):
+        multiarray_setop_validation(pda1, pda2)
+
+        if not assume_unique:
+            ag = GroupBy(pda1)
+            ua = ag.unique_keys
+            bg = GroupBy(pda2)
+            ub = bg.unique_keys
+        else:
+            ua = pda1
+            ub = pda2
+
+        # Key for deinterleaving result
+        isa = concatenate((ones(ua[0].size, dtype=akbool), zeros(ub[0].size, dtype=akbool)), ordered=False)
+        c = [concatenate(x, ordered=False) for x in zip(ua, ub)]
+        g = GroupBy(c)
+        if assume_unique:
+            # need to verify uniqueness, otherwise answer will be wrong
+            if (g.sum(isa)[1] > 1).any():
+                raise ValueError("Called with assume_unique=True, but first argument is not unique")
+            if (g.sum(~isa)[1] > 1).any():
+                raise ValueError("Called with assume_unique=True, but second argument is not unique")
+        k, ct = g.count()
+        truth = g.broadcast(ct == 1, permute=True)
+        return [x[truth[isa]] for x in ua]
```

Comment:
Similar to above, storing `truth[isa]` in a variable might save time and avoid recomputation inside the loop (again, not totally sure -- python might be smart enough to cache).

Reply:
To be safe, I updated the code to store `truth[isa]` to a var and call that in the loop.